cc-raisely-components/components/facil-conversation-list/facil-conversation-list.js

```diff
@@ -152,43 +235,148 @@
 
 		toggleFilter = () => {
 			// If we're about to switch to filtered
-			this.setState({	filter: !this.state.filter }, this.setConversations);
-		}
+			this.setState(
+				{ filter: !this.state.filter },
+				this.setConversations
+			);
+		};
+
+		toggleSelectConversations = () => {
+			//If we want to select duplicates to mergE
+			this.setState({
+				showCheckbox: !this.state.showCheckbox,
+			});
+		};
+
+		checkConversationAlreadyExists = (conversation) => {
+			const { selectedConversations } = this.state;
+
+			return selectedConversations.some(
+				(convo) => convo.uuid === conversation.uuid
+			);
+		};
+
+		setSelectedConversations = (conversation) => {
+			const { selectedConversations } = this.state;
+
+			if (this.checkConversationAlreadyExists(conversation)) {
+				const convo = selectedConversations.filter(
+					(convo) => convo.uuid !== conversation.uuid
```

Comment:
You should use a different variable in the filter function. It's confusing that it uses convo which is declared above.

Since you only ever use the `uuid` you might be better off just storing that in the `selectedConversations` array.
You'd be able to turn all of this into just a couple of lines which leave less room for error and more readability:

```
let newSelected;
if (selectedConversations.includes(conversation.uuid) {
  newSelected = selectedConversations.filter(uuid => uuid !== conversation.uuid);
} else {
  newSelected = [...selectedConversations, conversation.uuid];
}
this.setState({ selectedConversations: newSelected });
```